net/initiate.go

```diff
@@ -164,3 +181,16 @@ func (h *Host) handleProtocolStreamInner(stream libp2pnetwork.Stream, s SwapStat
 		}
 	}
 }
+
+func (h *Host) handleProtocolStreamClose(stream libp2pnetwork.Stream, s SwapState) {
+	log.Debugf("closing stream: peer=%s protocol=%s", stream.Conn().RemotePeer(), stream.Protocol())
+	_ = stream.Close()
+
+	log.Debugf("exiting swap...")
+	if err := s.Exit(); err != nil {
+		log.Errorf("failed to exit protocol: err=%s", err)
```

Comment:
We've been leaving the `err=` part of on newer messages. People already expect the error after the colon.